src/pages/UnlockedWallet/AddressesPage/AddressesTabContent.tsx

```diff
@@ -32,20 +35,74 @@ import AddressCard from './AddressCard'
 const AddressesTabContent = () => {
   const { addresses } = useAddressesContext()
   const { t } = useTranslation()
+  const newAddressButtonRef = useRef<HTMLDivElement>(null)
+  const isInView = useInView(newAddressButtonRef)
 
   const [isGenerateNewAddressModalOpen, setIsGenerateNewAddressModalOpen] = useState(false)
+  const [searchResults, setSearchResults] = useState(addresses)
+  const [emptyAddressesToggleResults, setEmptyAddressesToggleResults] = useState(addresses)
+  const [hideEmptyAddresses, setHideEmptyAddresses] = useState(false)
+
+  const visibleAddresses = searchResults.filter((address1) => emptyAddressesToggleResults.includes(address1))
+
+  const handleSearch = (searchInput: string) => {
+    if (searchInput.length < 2) {
+      setSearchResults(addresses)
```

Comment:
Will this trigger a re-render?

Reply:
Not if the value `searchResults` value remains the same.